cmd/diode/bns.go

```diff
@@ -121,49 +125,75 @@ func handleRegister() (done bool, err error) {
 		return
 	}
 
-	var obnsAddr util.Address
+	var obnsAddr []util.Address
 	// should lowercase bns name
 	bnsName := strings.ToLower(registerPair[0])
 	if !isValidBNS(bnsName) {
 		cfg.PrintError("Argument Error: ", fmt.Errorf("BNS name should be more than 7 or less than 32 characters (0-9A-Za-z-)"))
 		return
 	}
 	nonce := client.GetAccountNonce(0, cfg.ClientAddr)
-	var bnsAddr util.Address
+	var bnsAddr []util.Address
 	if len(registerPair) > 1 {
-		bnsAddr, err = util.DecodeAddress(registerPair[1])
-		if err != nil {
-			cfg.PrintError("Invalid diode address", err)
-			return
+		for _, strAddr := range strings.Split(registerPair[1], ",") {
+			var addr util.Address
+			addr, err = util.DecodeAddress(strAddr)
+			if err != nil {
+				cfg.PrintError("Invalid diode address", err)
+				return
+			}
+			bnsAddr = append(bnsAddr, addr)
 		}
 	} else {
-		bnsAddr = cfg.ClientAddr
+		bnsAddr = append(bnsAddr, cfg.ClientAddr)
 	}
 	// check bns
 	obnsAddr, err = client.ResolveBNS(bnsName)
-	if err == nil {
-		if obnsAddr == bnsAddr {
-			cfg.PrintError("BNS name is already mapped to this address", err)
+	if err == nil && len(obnsAddr) == len(bnsAddr) {
+		if util.Equal(obnsAddr, bnsAddr) && !cfg.BNSForce {
+			cfg.PrintError("BNS name is already mapped to this address", fmt.Errorf("ignored"))
 			return
 		}
 	}
+
 	// send register transaction
 	var res bool
 	registerData, _ := bnsContract.Register(bnsName, bnsAddr)
 	ntx := edge.NewTransaction(nonce, 0, 10000000, contract.BNSAddr, 0, registerData, 0)
 	res, err = client.SendTransaction(ntx)
+	if err == nil && !res {
+		err = fmt.Errorf("server returned false")
```

Comment:
How about return error from SendTransaction when response not equal to "ok"?

Reply:
Yes! I was thinking the same when i look at the code, but wasn't sure about other places where it is used.